x/concentrated-liquidity/fees.go

```diff
@@ -253,6 +253,51 @@ func (k Keeper) collectFees(ctx sdk.Context, poolId uint64, owner sdk.AccAddress
 	return feesClaimed, nil
 }
 
+// queryClaimableFees queries the fee accumulator for the position given by pool id, owner, lower tick and upper tick.
+// It returns the outstanding fees that can be claimed by the owner.
+// Returns error if:
+// - pool with the given id does not exist
+// - position given by pool id, owner, lower tick and upper tick does not exist
+// - other internal database or math errors.
+func (k Keeper) queryClaimableFees(ctx sdk.Context, poolId uint64, owner sdk.AccAddress, lowerTick int64, upperTick int64) (sdk.Coins, error) {
```

Comment:
Let's have a basic test making sure that this query does not modify state please

Reply:
Does the test I wrote not cover this? If the query mutated state, the amount we collect after calling the query would be different than expected

Reply:
What I was thinking of is having an explicit check that the position fee accumulator stays the same after the claim. I think this would be a tighter check than checking the collected values.

Reply:
Sounds good, I implemented this here https://github.com/osmosis-labs/osmosis/pull/4501/commits/5aacf7b8c0c8f8613055a93c6c2737e2786274ff

I had to export GetPosition in order to compare the init values, since that is what gets modified when we claim rewards. Additionally, to work with error cases (without making it hacky), I had to differentiate with a MustGetPosition vs GetPosition. Previously, GetPosition acted as a MustGet. So I replaced all current values of GetPosition with MustGet, and utilized the regular GetPosition for tests.